components/imagePicker/index.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable camelcase */
+import { __ } from '@wordpress/i18n';
+import { BaseControl, Button, ButtonGroup } from '@wordpress/components';
+import { useEffect, useRef, useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import type { WP_REST_API_Attachment } from 'wp-types';
+import './index.scss';
+
+type ImagePickerProps = {
+  label: string;
+  onChange: (value: number) => void;
+  value: number;
+};
+
+type MediaLibraryOptions = {
+  library: {
+    type: string;
+  };
+};
+
+type MediaLibrarySelection = {
+  id: number;
+  url: string;
+};
+
+declare global {
+  interface Window {
+    wp: {
+      media: (options: MediaLibraryOptions) => any;
+    };
+  }
+}
+
+export default function ImagePicker({
```

Comment:
We have an image picker component available as part of `block-editor-tools` which I would like to use here if possible: https://github.com/alleyinteractive/alley-scripts/tree/main/packages/block-editor-tools/src/components/image-picker